utils/DMTHandler.js

```diff
@@ -1,154 +1,193 @@
+/* eslint-disable class-methods-use-this */
 /* eslint-disable no-param-reassign */
 /* eslint-disable prefer-promise-reject-errors */
 /* eslint-disable consistent-return */
-import { replacer } from './utils';
+import { replacer, removeTrailingSlashes } from './utils';
 import logger from './logUtil';
 
-// default values for retry
-const retryCount = 3;
+class DMTHandler {
+  constructor(writeKey, dataPlaneUrl) {
+    this.dataPlaneUrl = removeTrailingSlashes(dataPlaneUrl);
+    this.writeKey = writeKey;
+    this.retryCount = 3; // default value for retry
+    this.queue = [];
+    this.isTransformationProcessing = false;
+    this.isTimerStarted = false;
+  }
 
-/**
- * A helper function that will take rudderEelement as an event and generate
- * a batch payload that will be sent to transformation server
- *
- */
-const createPayload = (event) => {
-  const orderNo = Date.now();
-  const payload = {
-    batch: [
-      {
-        orderNo,
-        event,
-      },
-    ],
-  };
-
-  return payload;
-};
+  // Enqueue the events and callbacks
+  enqueue(event, cb) {
+    this.queue.push({
+      event,
+      cb,
+    });
+    this.start();
+  }
 
-/**
- * A helper function that will take one payload at a time and send it to transformation server
- * and return the response
- *
- */
-const sendEventForTransformation = (payload, writeKey, dataPlaneUrl, retryCount) => {
-  return new Promise((resolve, reject) => {
-    // const url = `${dataPlaneUrl}/transform`;
-    const url = 'https://shadowfax-dataplane.dev-rudder.rudderlabs.com/transform';
-    const headers = {
-      'Content-Type': 'application/json',
-      Authorization: `Basic ${btoa(`${writeKey}:`)}`,
+  /**
+   * A helper function that will take rudderEelement as an event and generate
+   * a batch payload that will be sent to transformation server
+   *
+   */
+  createPayload(event) {
```

Comment:
This doesn't have to be a class method.

Reply:
As this method will only be used in DMT let's keep it here.

Reply:
Let's take the function out and define it along with the class or move into DMT utils.

Reply:
Done.